sdk/src/sharedTest/java/io/customer/sdk/queue/QueueRunRequestTest.kt

```diff
@@ -67,41 +74,109 @@ class QueueRunRequestTest : BaseTest() {
     }
 
     @Test
-    fun test_start_givenHttpRequestsPaused_expectDontDeleteTask_expectDontUpdateTask(): Unit = runBlocking {
+    fun test_run_givenRunTask400Failure_expectDeleteTask(): Unit = runBlocking {
```

Comment:
Do we have integration tests setup for the Queue? Aka: tests that do not require so much mocking such as `storageMock`? 

I think the tests you added are testing good edge cases but they could be higher quality and less annoying to maintain (since we don't use as many mocks) if they were integration tests.